test/util/blobfactory/payforblob_factory.go

```diff
@@ -23,6 +24,13 @@ import (
 
 var defaultSigner = testfactory.RandomAddress().String()
 
+var (
+	// TestMaxBlobSize is the maximum size of each blob in a blob transaction, for testing purposes
+	TestMaxBlobSize = appconsts.ShareSize * 4
```

Comment:
When setting sizes larger than this, specifically close to the square row size, the fuzzer exhibits non-deterministic halting behavior. I have also verified this issue by testing the original square fuzzer in the main branch, and encountered the same problem. I am currently investigating this issue and will submit another pull request as soon as I identify a fix.

Reply:
Tracking issue https://github.com/celestiaorg/celestia-app/issues/1956

Reply:
:eyes: :eyes: 

fwiw, we are testing with blobs larger than this in the integration and square size test, but this does seem like a very serious bug!

https://github.com/celestiaorg/celestia-app/blob/c1176b18ada45a2465cb48dc48eface32d301fe3/app/test/integration_test.go#L79-L107

Reply:
Thanks, @EvanForbes, for your input! It appears that the issue I encountered was specific to my local machine. I tried running it on another machine, and it worked perfectly fine. Therefore, I'm going to revert the max blob size back to the larger value, which is two rows of a square. https://github.com/celestiaorg/celestia-app/pull/1860/commits/087d8addcc1ad4edcebbcfbd8ff18a85270c6f99